openpype/hosts/fusion/plugins/publish/collect_instances.py

```diff
@@ -24,23 +22,39 @@ def process(self, instance):
         creator_attributes = instance.data["creator_attributes"]
         instance.data.update(creator_attributes)
 
-        # Include start and end render frame in label
-        subset = instance.data["subset"]
+        # get asset frame ranges
         start = context.data["frameStart"]
         end = context.data["frameEnd"]
+        handle_start = context.data["handleStart"]
+        handle_end = context.data["handleEnd"]
+        start_handle = start - handle_start
+        end_handle = end + handle_end
+
+        if creator_attributes.get("custom_range"):
+            # get comp frame ranges
+            start = context.data["compFrameStart"]
+            end = context.data["compFrameEnd"]
+            handle_start = 0
+            handle_end = 0
+            start_handle = start
+            end_handle = end
```

Comment:
This is now using custom frame range defined on viewer timeline. Once render range is defined and the instance attribute `custom range` is enabled then Render frame range is used and handles needs to be at 0.

Reply:
>  and handles needs to be at 0.

Why is this the case? Didn't we explicitly add at some point that the global in/out would be the handles for the render range in/out? Why can't we use that as the handles for the comp's range?

Reply:
> > and handles needs to be at 0.
> 
> Why is this the case? Didn't we explicitly add at some point that the global in/out would be the handles for the render range in/out? Why can't we use that as the handles for the comp's range?

I would be quite confused by that without distinct information that that's the case.

Reply:
> I would be quite confused by that without distinct information that that's the case.

By what? That it's used, or that it's not? What's your expected behavior? (I personally don't care too much about handles since we don't use it much from comps in our productions - just wondering what the reasoning is here)

Reply:
> > I would be quite confused by that without distinct information that that's the case.
> 
> By what? That it's used, or that it's not? What's your expected behavior? (I personally don't care too much about handles since we don't use it much from comps in our productions - just wondering what the reasoning is here)

If I select custom range that will render my in/out points in Fusion I would be confused why it starts to render the whole thing (if I didn't know i had to also set the global in/out for the handles-part)

Reply:
I still don't know whether "Custom" means take the one from comp or not - haha. Anyway, [using this enum definition of options](https://github.com/ynput/OpenPype/pull/4955#discussion_r1192263496):

1. Use comp range
2. Use asset frame range
3. Use custom instance frame range

You're saying that 1) using the comp's range should NEVER include handles and ever render the 'render range' (not global in/global out)? Or should it always render the full global in/global out as the frame range?

Reply:
For me `comp range` would mean the global in/out.
I would suggest changing the name to `render range` and use the render in/out as that would be the most useful option

Reply:
> For me `comp range` would mean the global in/out. I would suggest changing the name to `render range` and use the render in/out as that would be the most useful option

Only reason `comp` had been added as prefix here was that it was collected in `CollectFusionCompFrameRanges` and it is referring to comp object as source of frame range. But I don't have a problem to rename it to render ;).

Reply:
> > and handles needs to be at 0.
> 
> Why is this the case? Didn't we explicitly add at some point that the global in/out would be the handles for the render range in/out? Why can't we use that as the handles for the comp's range?

Let me verify this by looking into the behavior before this PR. The Handles were at 0 if viewer render range were not set (therefore at -1000000000 value). So how did you set the handles then? Or was that a problem for you before?